app/sagas/troubleshootingNotification.ts

```diff
@@ -0,0 +1,45 @@
+import { Action } from 'redux';
+import { put, takeEvery } from 'redux-saga/effects';
+import { call } from 'typed-redux-saga';
+import notifee from '@notifee/react-native';
+
+import { TROUBLESHOOTING_NOTIFICATION } from '../actions/actionsTypes';
+import { setTroubleshootingNotification } from '../actions/troubleshootingNotification';
+import { pushInfo } from '../lib/services/restApi';
+import log from '../lib/methods/helpers/log';
+import { appSelector } from '../lib/hooks';
+import { compareServerVersion } from '../lib/methods/helpers';
+
+interface IGenericAction extends Action {
+	type: string;
+}
+
+function* request() {
+	const serverVersion = yield* appSelector(state => state.server.version);
+	let deviceNotificationEnabled = false;
+	let defaultPushGateway = false;
+	let pushGatewayEnabled = false;
+	try {
+		const { authorizationStatus } = yield* call(notifee.getNotificationSettings);
+		deviceNotificationEnabled = authorizationStatus > 0;
+		const pushInfoResult = yield* call(pushInfo);
+		if (pushInfoResult.success) {
+			pushGatewayEnabled = pushInfoResult.pushGatewayEnabled;
+			defaultPushGateway = pushInfoResult.defaultPushGateway;
+		}
+	} catch (e) {
+		log(e);
+	} finally {
+		// If Any of the items that can have red values: notification settings, CE quota, or gateway connection; the red icon should show.
+		// Then inAlertNotification has to be true
+		const inAlertNotification =
```

Comment:
`IN` as `inAlertNotification` sounds more like a physical place, like a modal, than a state, which I think is what you meant to do.
This state is used to determine the danger level of the push notifications.
You could call it `highlight: boolean` or `enabled`, which would be used as `troubleshootingNotifications.highlight`.
I still think we shouldn't display the buttons every time though, so we could end up transforming this state into `troubleshooting: boolean` (or `showTroubleshootingButton`).

Reply:
I thought in use `highlightTroubleshooting`, what do you think?